amr-wind/equation_systems/icns/source_terms/BodyForce.cpp

```diff
@@ -0,0 +1,60 @@
+#include "amr-wind/equation_systems/icns/source_terms/BodyForce.H"
+#include "amr-wind/CFDSim.H"
+#include "amr-wind/utilities/trig_ops.H"
+
+#include "AMReX_ParmParse.H"
+#include "AMReX_Gpu.H"
+
+namespace amr_wind {
+namespace pde {
+namespace icns {
+
+/** Body Force
+ */
+BodyForce::BodyForce(const CFDSim& sim)
+    : m_time(sim.time()), m_density(sim.repo().get_field("density"))
+{
+
+    {
+        // Read the geostrophic wind speed vector (in m/s)
+        amrex::ParmParse pp("BodyForce");
+        pp.query("type", m_type);
+        pp.queryarr("magnitude", m_body_force);
+        if (m_type == "Oscillatory" || m_type == "oscillatory") {
+            pp.query("AngularFrequency", m_omega);
```

Comment:
We have been using camel case to indicate runtime selection of sub-classes and snake case for input variables. Your choice if you want to use camel case for variable names.